lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1121,6 +1125,26 @@ sub _create_bam_flagstat {
     return 1;
 }
 
+sub create_bam_header {
+    my $self = shift;
+
+    if (-s $self->bam_header_path) {
+        return 1;
+    }
+
+    my $lock = $self->get_bam_lock;
```

Comment:
If there are multiple return points (or exceptions) then `Scope::Guard` may be handy:

```
    my $lock = $self->get_bam_lock();
    my $guard = Scope::Guard->new(sub { $lock->unlock() });
```

Which would unlock the lock as soon as `$guard` goes out of scope which means on any return or any exception.